sympy/solvers/tests/test_solvers.py

```diff
@@ -356,7 +356,7 @@ def test_solve_transcendental():
     assert set(solve((a*x + b)*(exp(x) - 3), x)) == set([-b/a, log(3)])
     assert solve(cos(x) - y, x) == [-acos(y) + 2*pi, acos(y)]
     assert solve(2*cos(x) - y, x) == [-acos(y/2) + 2*pi, acos(y/2)]
-    assert solve(Eq(cos(x), sin(x)), x) == [pi*Rational(-3, 4), pi/4]
+    assert solve(Eq(cos(x), sin(x)), x) == [pi/4]
```

Comment:
Why has this changed?

I've never been clear exactly what should be the "correct" output from solve for this sort of thing.

Reply:
Periodic functions are not handled robustly like in solveset; on the other hand anyone solving a periodic function should know that there will be other solutions and they can easily generalize. In this case the period of `cos(x) - sin(x)` is `2pi` and the arg is `x` so there is another root at `pi + pi/4`.

Reply:
Perhaps it would be possible to add in the other solution when the periodicity of the original equation and the rewritten equation is not the same.

Reply:
Okay, that makes sense: the solutions are intended to be limited to the period of the original equation. If the equation is pi-periodic then the solutions should be enough to generate the full set under pi-periodicity.